Fundamental_Assignment_day2/HexCalc.java

```diff
@@ -0,0 +1,413 @@
+		String ResultString=ToHexadecimalNumber(result,16);
+		return ResultString;
+	}
+	public String Subtract(String first_number,String second_number)
+	{
+
+		/*
+		 * method for subtracting two hexadecimal numbers
+		 * @param decimal_first contains integer part of decimal value
+		 * @param decimal_second contains fraction part of decimal value
+		 * @param result subtracted value of decimal numbers
+		 * @param result_string contains resulting hexadecimal value in string
+		 * @return result_string
+		 */
+		float DecimalFirst= ToDecimal(first_number,16);
+		float DecimalSecond= ToDecimal(second_number,16);
+		float result;
+		if(DecimalFirst>DecimalSecond)
+		{
+			result= DecimalFirst-DecimalSecond;
+		}
+		else
+		{
+			result=DecimalSecond-DecimalFirst;
+		}
+		
+		String ResultString=ToHexadecimalNumber(result,16);
+		return ResultString;
+	}
+	public String Multiplication(String first_number,String second_number)
+	{
+
+		/*
+		 * method for multiplication of  two hexadecimal numbers
+		 * @param decimal_first contains integer part of decimal value
+		 * @param decimal_second contains fraction part of decimal value
+		 * @param result multiplied value of decimal numbers
+		 * @param result_string contains resulting hexadecimal value in string
+		 * @return result_string
+		 */
+		float DecimalFirst= ToDecimal(first_number,16);
+		float DecimalSecond= ToDecimal(second_number,16);
+		float result= DecimalFirst*DecimalSecond;
+		String ResultString=ToHexadecimalNumber(result,16);
+		return ResultString;	}
+	public String Divide(String first_number,String second_number)
+	{
+
+		/*
+		 * method for division of  two hexadecimal numbers
+		 * @param decimal_first contains integer part of decimal value
+		 * @param decimal_second contains fraction part of decimal value
+		 * @param result divided value of decimal numbers
+		 * @param result_string contains resulting hexadecimal value in string
+		 * @return result_string
+		 */
+		float DecimalFirst= ToDecimal(first_number,16);
+		float DecimalSecond= ToDecimal(second_number,16);
+		float result= DecimalFirst/DecimalSecond;
+		String ResultString=ToHexadecimalNumber(result,16);
+		return ResultString;
+	}
+
+	public boolean CheckHexadecimalNumber(String HexNumber)
+	{
+		/*
+		 * method for checking whether given number is in hexadecimal format or not
+		 * @param String hex_number
+		 * @ result boolean value
+		 */
+		char[] letters = HexNumber.toCharArray();
+		for (char itr: letters)
+		{
+			if(!((itr>='A' && itr<='F' )||( itr>='0' && itr<='9')|| itr=='.'))
+			{
+				return false;
+			}
+		}
+		return true;
+		
+	}
+	public boolean Compare(String FirstNumber, String SecondNumber)
+	{
+		/*
+		 * method for comparing two hexadecimal numbers 
+		 */
+		if(FirstNumber.length()<SecondNumber.length())
+		{
+			return true ;
+		}
+		else if(FirstNumber.length()>SecondNumber.length())
+		{
+			return false ;
+		}
+		else
+		{
+			for(int itr=0; itr<SecondNumber.length(); itr++)
```

Comment:
This is not correct logic for comparison.

E.g. "ABC" > "ADE" and "ADE" > "ABC"

Reply:
It is not working for IsEqual case, for other working fine.